app/web/src/store/secrets.store.ts

```diff
@@ -130,6 +135,49 @@ export function useSecretsStore() {
             },
           );
         },
+        secretsByLastCreated(state): SecretsOrderedArray {
+          const out = [] as SecretsOrderedArray;
+
+          for (const [key, value] of Object.entries(
+            state.secretDefinitionByDefinitionId,
+          )) {
+            let change: string | undefined;
+            value.secrets.forEach((secret) => {
+              if (!change) change = secret.createdInfo.timestamp;
+              if (new Date(secret.createdInfo.timestamp) > new Date(change)) {
+                change = secret.createdInfo.timestamp;
+              }
+              if (
+                secret.updatedInfo &&
+                new Date(secret.updatedInfo.timestamp) > new Date(change)
+              ) {
+                change = secret.updatedInfo.timestamp;
+              }
+            });
+
+            out.push({
+              id: key,
+              latestChange: change,
+              secrets: value.secrets,
+            });
+          }
+
+          out.sort((a, b) => {
+            if (!a.latestChange && !b.latestChange) return 0;
+            else if (!a.latestChange && b.latestChange) return 1;
+            else if (a.latestChange && !b.latestChange) return -1;
+            else if (a.latestChange && b.latestChange) {
```

Comment:
All of the previous cases returning values mean the code will only get here even without this if. This could be written something like:

```js
out.sort((a, b) => {
  if (!a.latestChange && !b.latestChange) return 0;
  if (!a.latestChange && b.latestChange) return 1;
  if (a.latestChange && !b.latestChange) return -1;

  return new Date(b.latestChange) - new Date(a.latestChange);
});
```

Also, Array.sort expects positive or negative values instead of only 1 and -1, so you could use subtraction of the dates to get a similar result. that's the reason for that last return line. See if you  like it

Reply:
Ah yeah - I figured there was a simpler way to do this, but I kept getting Typescript issues - even the solution you suggested results in a Typescript error, probably because TS is not smart enough to notice that the three if statements filter out any undefined values before that last return statement. 😑